### ISSUE TITLE
Section about %setup and %patch

### ISSUE THREAD

**Issue by xsuchy** [redhat-developer/rpm-packaging-guide#20]
Can you please add a section about %setup and %patch. Those are quite important macros.

I think that you can copy and paste http://ftp.rpm.org/max-rpm/s1-rpm-inside-macros.html

**Comment by Daskyleia** [comment]
We cannot copy paste a text from upstream sources. It should be taken only as an inspiration to write our own.

I need to have a use case for %patch. In our guide, there is a chapter about patching the cello.c code, maybe we can somehow use it there? Please, send me anything you think would be a good use case scenario for our guide.

**Comment by xsuchy** [comment]
> We cannot copy paste a text from upstream sources.

@Daskyleia  RPM Max is (c) by Red Hat. There is nothing holding you back.

**Comment by Daskyleia** [comment]
Yes, but the document should not be a reference guide, we should make some use case scenario...
